src/core/radio/ble_secure.cpp

```diff
@@ -0,0 +1,542 @@
+
+        if (mTcatAgent.IsEnabled())
+            mTcatAgent.Disconnected();
+    }
+
+    mConnectCallback.InvokeIfSet(&GetInstance(), aConnected, true);
+}
+
+void BleSecure::HandleTlsReceive(void *aContext, uint8_t *aBuf, uint16_t aLength)
+{
+    return static_cast<BleSecure *>(aContext)->HandleTlsReceive(aBuf, aLength);
+}
+
+void BleSecure::HandleTlsReceive(uint8_t *aBuf, uint16_t aLength)
+{
+    VerifyOrExit(mReceivedMessage != nullptr);
+
+    if (!mTlvMode)
+    {
+        SuccessOrExit(mReceivedMessage->AppendBytes(aBuf, aLength));
+        mReceiveCallback.InvokeIfSet(&GetInstance(), mReceivedMessage, 0, OT_TCAT_MESSAGE_TYPE_NONE, "");
+        IgnoreReturnValue(mReceivedMessage->SetLength(0));
+    }
+    else
+    {
+        ot::Tlv  tlv;
+        uint32_t requiredBytes = sizeof(Tlv);
+        uint32_t offset;
+
+        while (aLength > 0)
+        {
+            if (mReceivedMessage->GetLength() < requiredBytes)
+            {
+                uint32_t missingBytes = requiredBytes - mReceivedMessage->GetLength();
+
+                if (missingBytes > aLength)
+                {
+                    SuccessOrExit(mReceivedMessage->AppendBytes(aBuf, aLength));
+                    break;
+                }
+                else
+                {
+                    SuccessOrExit(mReceivedMessage->AppendBytes(aBuf, (uint16_t)missingBytes));
+                    aLength -= missingBytes;
+                    aBuf += missingBytes;
+                }
+            }
+
+            IgnoreReturnValue(mReceivedMessage->Read(0, tlv));
+
+            if (tlv.IsExtended())
+            {
+                ot::ExtendedTlv extTlv;
+                requiredBytes = sizeof(extTlv);
+
+                if (mReceivedMessage->GetLength() < requiredBytes)
+                {
+                    continue;
+                }
+
+                IgnoreReturnValue(mReceivedMessage->Read(0, extTlv));
+                requiredBytes = extTlv.GetSize();
+                offset        = sizeof(extTlv);
+            }
+            else
+            {
+                requiredBytes = tlv.GetSize();
+                offset        = sizeof(tlv);
+            }
+
+            if (mReceivedMessage->GetLength() < requiredBytes)
+            {
+                continue;
+            }
+
+            // TLV fully loaded
+
+            if (mTcatAgent.IsEnabled())
+            {
+                ot::Message *message;
+                Error        error = kErrorNone;
+
+                message = Get<MessagePool>().Allocate(Message::kTypeBle);
+                VerifyOrExit(message != nullptr, error = kErrorNoBufs);
+
+                error = mTcatAgent.HandleSingleTlv(*mReceivedMessage, *message);
+                if (message->GetLength() != 0)
+                {
+                    IgnoreReturnValue(SendMessage(*message));
+                }
+
+                if (error == kErrorAbort)
+                {
+                    Disconnect(); // Keep BLE active (for testing)
```

Comment:
comment is unclear - BLE is stopped in the next line?

Reply:
Removed the comment. and Yes BLE is stopped in next line.